### ISSUE TITLE
core_shell_fractal uses wrong effective radius for the fractal structure factor (Trac #792)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#894]
The `core_shell_fractal` model uses `radius` as the effective radius for the structure rather than `radius+thickness`; this cannot be right. For example, if you were to have a core shell with a very thick shell and almost no core, the particles could not get within the core radius of each other.

Migrated from http://trac.sasview.org/ticket/792
```json
{
    "status": "closed",
    "changetime": "2016-11-28T07:01:07",
    "_ts": "2016-11-28 07:01:07.019326+00:00",
    "description": "The `core_shell_fractal` model uses `radius` as the effective radius for the structure rather than `radius+thickness`; this cannot be right. For example, if you were to have a core shell with a very thick shell and almost no core, the particles could not get within the core radius of each other.",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-10-20T14:20:05",
    "component": "SasView",
    "summary": "core_shell_fractal uses wrong effective radius for the fractal structure factor",
    "priority": "minor",
    "keywords": "",
    "milestone": "SasView 4.1.0",
    "owner": "butler",
    "type": "defect"
}
```

**Comment by pkienzle** [comment]
Trac update at `2016/11/08 14:54:56`: 
* **pkienzle** changed owner from "" to "butler"
* **pkienzle** changed status from "new" to "assigned"

**Comment by butlerpd** [comment]
Trac update at `2016/11/28 06:02:45`: **butler** commented:

That is correct.  The Original IGOR code does it correctly as `radius+thickness` but seems to have been translated incorrectly as that does not seem to be in libigor.  Assuming the translation was done from the documentation which for the core shell only said see the parent documentation (fractal, and core_shell) andof course fractal only has one r?

At any rate it is obvious if one uses say 60 radius and 10 core with a core sld=shellsld then swapping so that radius = 10 and shell = 60 cannot possibly change the curve... but does.

The fix however cannot be checked right now since there is a deeper problem which has been introduced since 4.01 which is that for openCL on my windows 10 the current build passes 0 for thickness to the model.  It seems to pass the correct value for my windows 10 using DLL however.  I am opening another ticket to deal with that first.  In the meantime the documentation of the model has been cleaned up and the equation verified to match the what will be the calculation.

**Comment by butlerpd** [comment]
Trac update at `2016/11/28 06:54:25`: 
* **butler** commented:

In changeset bdd08df2ecb671a5503c0699fe24a2bc3f2a8a7a:

```text
#!CommitTicketReference repository="sasmodels" revision="bdd08df2ecb671a5503c0699fe24a2bc3f2a8a7a"
corrects fractal core shell S(q) to be based on radius of full particle
(core radius + thickness) and tested.  Fixes #894
```


* **butler** changed resolution from "" to "fixed"
* **butler** changed status from "assigned" to "closed"

**Comment by butlerpd** [comment]
Trac update at `2016/11/28 07:01:07`: **butler** commented:

Actually it of course can be checked by using opencl = False.  Tested with original code and showed that with sld core = sld shell which means in fact a large uniform sphere, and putting 60A for the core radius and 10A for the shell the scattering must be unchanged when swapping the values so that core is 10A and shell is 60.  In fact there is a dramatic change.  With the fix that makes sense from the math and is conistant with the original IGOR code, the scattering is now indeed invariant to that change.  Further by changing the SLD of the shell to be between the core and solvent yields a small shift in intensity AND a slight shift in the minima of the oscillations as it should.  Thus I consider this ticket fixed but have opened another regarding the opencl problem.
